Stratis.Bitcoin.Tests/BlockStore/LoopTests/BlockStoreLoopIntegration.cs

```diff
@@ -0,0 +1,160 @@
+﻿using NBitcoin;
+using Stratis.Bitcoin.Features.BlockStore.LoopSteps;
+using System.Linq;
+using System.Threading;
+using Xunit;
+using static Stratis.Bitcoin.BlockPulling.BlockPuller;
+
+namespace Stratis.Bitcoin.Tests.BlockStore.LoopTests
+{
+    public sealed class BlockStoreLoopIntegration : BlockStoreLoopStepBaseTest
+    {
+        /// <summary>
+        /// Tests the block store loop step with a concrete implementation of BlockRepository.
+        /// </summary>
+        [Fact]
+        public void CheckNextChainedBlockExists_Integration()
```

Comment:
Very nice.

In regards to method names can you follow what @bokobza did in the wallet?
Given_Xxx_When_Xxx_Then_Xxx()
Given_Xxx_AndGiven_Xxx_When_Xxx_AndWhen_Xxx_Then_Xxx_AndThen_Xxx()

We are trying out several test naming conventions and when we hit the right convention we will replace across the board.